dlt/common/schema/annotations.py

```diff
@@ -0,0 +1,162 @@
+import inspect
+
+from typing import Type, List, Any, Tuple, Union, Dict
+from typing_extensions import Annotated, get_origin, get_args
+from decimal import Decimal
+
+from dataclasses import dataclass
+from dlt.common.schema.typing import TTableSchema, TColumnSchema, TWriteDisposition, TDataType
+from datetime import datetime, date
+
+
+#
+# base dataclasses used for hints
+#
+@dataclass
+class BoolValue:
+    value: bool
+
+
+@dataclass
+class StringValue:
+    value: str
+
+
+@dataclass
+class IntValue:
+    value: int
+
+
+@dataclass
+class StrListValue:
+    value: List[str]
+
+
+#
+# Column Hints
+#
+class PrimaryKey(BoolValue): ...
+
+
+class Unique(BoolValue): ...
+
+
+#
+# Table Hints
+#
+class TableName(StringValue): ...
+
+
+class Description(StringValue): ...
+
+
+class Classifiers(StringValue): ...
+
+
+@dataclass
+class WriteDisposition:
+    value: TWriteDisposition
+
+
+#
+# Converters
+#
+
+TypeMap: Dict[Any, TDataType] = {
+    str: "text",
```

Comment:
we have py_type_to_sc_type. look at the type_helpers.py. tons of edge cases are handled there when converting types

Reply:
ah yes, great :) I was wondering if we had something like this somewhere.